packages/interface/README.md

```diff
@@ -0,0 +1,77 @@
+# (Experimental) Interface
+
+The Interface Package contains the basis the start a modern WordPress screen as "core/edit-post" and "core/edit-site". The package offers a store and a set of components. The store is useful to contain common data required by a screen (e.g., active areas). The information is persisted across screen reloads. The components allow one to implement functionality like a sidebar and allow a screen to be extended by third-party plugins by default.
+
+
+
+## Installation
+
+Install the module
+
+```bash
+npm install @wordpress/interface --save
+```
+
+_This package assumes that your code will run in an **ES2015+** environment. If you're using an environment that has limited or no support for ES2015+ such as lower versions of IE then using [core-js](https://github.com/zloirock/core-js) or [@babel/polyfill](https://babeljs.io/docs/en/next/babel-polyfill) will add support for these methods. Learn more about it in [Babel docs](https://babeljs.io/docs/en/next/caveats)._
+
+
+### API Usage
+
+
+
+#### Store
```

Comment:
Reading the README it's unclear to me what this store is about at this point, I'm introduced to something called areas (multiple and single) but I don't know what these areas are at this point.

Reply:
Technical remark: Adding a store to a package make that store a singleton, thus a bit less reusable. I'd have preferred if the API was only based on components but I'm not sure how possible is that.

Reply:
> I'd have preferred if the API was only based on components but I'm not sure how possible is that.

We can have a provider that keeps state passes it by context and the ComplementaryArea and PinnedItems components access that state.
There are some challenges with that approach:
	- As it is not a store we would not automatically use the persistent mechanisms a store provides and we would would need to implement something new.
	- Currently to open/close sidebars third-party plugins can dispatch actions to edit-post (our code also uses that). We totally removed the reducers and store logic for edit-post sidebars but we are being back-compatible with all the edit-post store actions and selectors. If we used a components approach without a store being back-compatible would be more tricky.

Reply:
It'd be interesting to explore persistent contexts :)

Reply:
> It'd be interesting to explore persistent contexts :)

That's is something we could do, the more tricky part would be the second point: being back-compatible with the current store calls.
The provider for this area would also need to be above the plugin content provider and we would need to provide an API probably context-based that plugins can use to open complementary areas in new screens (where they can not use the edit-post back-compatibility layer).

Reply:
Seems like a bit too much to replicate, I wonder if at some point we could move away from Global stores and just have "local stores" (which is basically the same thing as a context)